### PR TITLE
Add gradient for sqrt function

### PR DESCRIPTION
**Pull request by erikabor**
I added the gradient for the `sqrt` function. Please check if the implementation is right.

### CODE REVIEW

File: tensorflow-ops/src/TensorFlow/Gradient.hs [hunk 1]
```diff
@@ -813,6 +813,10 @@ opGrad "Placeholder" _ _ _ = []
 opGrad "VarHandleOp" _ _ _ = []
 opGrad "Variable" _ _ _ = []
 
+opGrad "Sqrt" _ [toT -> x] [dz] = [Just $ sq' `CoreOps.mul` dz]
+  where
+    sq' = vector [1] `CoreOps.div` (vector [2] `CoreOps.mul` CoreOps.sqrt x)
```
Comment (blackgnezdo) [inline tensorflow-ops/src/TensorFlow/Gradient.hs]: I don't remember, is there a way to multiply by a scalar rather than a (implicitly?) replicated vector?

Comment (fkm3) [reply tensorflow-ops/src/TensorFlow/Gradient.hs]: I think

```haskell
scalar 1 `CoreOps.div` (scalar 2 `CoreOps.mul` CoreOps.sqrt x)
```

would work,  but if that doesn't compile, then the proposed implementation is fine.

File: tensorflow-ops/tests/GradientTest.hs [hunk 2]
```diff
@@ -324,6 +324,14 @@ testPad =
     V.fromList [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] @=? dx
     V.fromList [2, 2, 3] @=? s
 
+testSqrt :: Test
+testSqrt = testCase "testSqrt" $ do
+    [dx] <- TF.runSession $ do
+        x <- TF.render $ TF.vector [0.25 :: Float]
```
Comment (fkm3) [inline tensorflow-ops/tests/GradientTest.hs]: Could you choose an input where the expected gradient isn't 1 or 0?

File: tensorflow-ops/src/TensorFlow/Gradient.hs (hunk 1 above)
Comment (erikabor) [reply tensorflow-ops/src/TensorFlow/Gradient.hs]: Yes, the multiplication with a scalar works. It compiles.

File: tensorflow-ops/tests/GradientTest.hs (hunk 2 above)
Comment (erikabor) [reply tensorflow-ops/tests/GradientTest.hs]: Yes, I've just fixed it.

**Pull request merged**